### PR TITLE
MiniCssExtractPlugin setup

### PR DESCRIPTION
**Pull request by rob-balfre**
https://github.com/sveltejs/sapper-template/pull/31#issuecomment-398594456

### CODE REVIEW

File: webpack/client.config.js [hunk 1]
```diff
@@ -20,14 +22,26 @@ module.exports = {
...
+					"css-loader"
+				]
 			}
 		]
 	},
 	mode,
 	plugins: [
+		new MiniCssExtractPlugin({
+			filename: "[name].css",
+			chunkFilename: "[id].css"
```
Comment (MaxMilton) [inline webpack/client.config.js]: The naming should probably match the sapper JS path and file name, so something like:
```
new MiniCssExtractPlugin({
      filename: '[hash]/[name].css',
      chunkFilename: '[hash]/[name].[id].css'
    }),
```

File: webpack/client.config.js [hunk 2]
```diff
@@ -20,14 +22,26 @@ module.exports = {
...
-						hotReload: true
+						hotReload: true,
+						emitCss: true
 					}
 				}
+			},
+			{
+				test: /\.css$/,
+				use: [
+					isDev ? 'style-loader' : MiniCssExtractPlugin.loader,
```
Comment (MaxMilton) [inline webpack/client.config.js]: This doesn't take into consideration source maps. That's probably something that should be out of the box.

Maybe something like:
```
use: [
          isDev
            ? { loader: 'style-loader', options: { sourceMap: true }}
            : MiniCssExtractPlugin.loader,
          { loader: 'css-loader', options: { sourceMap: isDev }}
        ],
```

File: webpack/client.config.js (hunk 1 above)
Comment (MaxMilton) [reply webpack/client.config.js]: There's also the issue of how to inject a reference to this file in the HTML template. I created an issue for that a while ago: https://github.com/sveltejs/sapper-template/issues/59

**Comment by Rich-Harris** [comment]
This is fantastic, thank you! I agree with @MaxMilton that it should probably be `[hash]/[name].css` and `[hash]/[id].css`, and the sourcemap point is a good one (that's currently broken anyway; will fix separately).

I won't merge this just yet because there's an annoying bug that needs fixing in Sapper first --> https://github.com/sveltejs/sapper/issues/116

**Comment by nolanlawson** [comment]
How does this play with the injection of critical CSS? In this PR the critical CSS seems to be injected just fine, but when I use the same configuration, I get a flash of unstyled content because there's no longer any critical CSS (https://github.com/nolanlawson/pinafore/issues/798). Maybe it has something to do with the subtle interplay with async chunks?

**Pull request closed without merging**